go/vt/binlog/binlogplayer/mock_dbclient.go

```diff
@@ -57,6 +58,10 @@ func NewMockDBClient(t *testing.T) *MockDBClient {
 			"select id, type, state, message from _vt.vreplication_log": {},
 			"insert into _vt.vreplication_log":                          {},
 		},
+		// For some reason the following queries are not sent in a determinstic order.
+		ignored: map[string]struct{}{
+			"set foreign_key_checks=1": {},
+		},
```

Comment:
This addresses the flakiness seen in the unit57 workflow. For some reason the order of this command was not deterministic in that workflow.